test/tools/ossfuzz/protoToYul.h

```diff
@@ -295,12 +309,16 @@ class ProtoConverter
 	}
 
 	std::ostringstream m_output;
-	/// Variables in current scope
-	std::stack<std::vector<std::string>> m_scopeVars;
+	/// Variables in all function definitions
+	std::vector<std::vector<std::vector<std::string>>> m_funcVars;
```

Comment:
Why a 3d vector instead of 2d? The way I assumed this would be is:
1- vector of function scopes, where
2- each element is a vector of variables

Reply:
1- vector of functions, where
2- each element is a vector of scopes within indexed function, where
3- each element is a vector of variables

Example:

```
function foo() { {let x} let y }

function bar() {let z}
```

creates a 3d vector like so {{{y}, {x}}, {{z}}}

Reply:
Right, but do you need to keep all scopes? When you visit `y`, `x`'s scope is irrelevant, isn't it?

Reply:
Ok, I think I got it now.

Reply:
Yeah, I only keep it till it is needed. When scope ends, that vector is popped.